src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorsAndIndexer/OperatorCompletionProvider.cs

```diff
@@ -0,0 +1,179 @@
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Data.SqlTypes;
+using System.Linq;
+using System.Numerics;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.Completion.Providers;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Completion.Providers
+{
+    [ExportCompletionProvider(nameof(OperatorCompletionProvider), LanguageNames.CSharp), Shared]
+    [ExtensionOrder(After = nameof(IndexerCompletionProvider))]
+    internal class OperatorCompletionProvider : OperatorIndexerCompletionProviderBase
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public OperatorCompletionProvider()
+        {
+        }
+
+        protected override IEnumerable<CompletionItem> GetCompletionItemsForTypeSymbol(ITypeSymbol container, SemanticModel semanticModel, int position)
+        {
+            if (!IsExcludedFromOperators(semanticModel, container))
+            {
+                var allMembers = container.GetMembers();
+                var operators = from m in allMembers.OfType<IMethodSymbol>()
+                                where m.IsUserDefinedOperator() && !IsExcludedOperator(m)
+                                let sign = m.GetOperatorSignOfOperator()
+                                select SymbolCompletionItem.CreateWithSymbolId(
+                                    displayText: sign,
+                                    displayTextSuffix: m.GetOperatorName(),
+                                    filterText: "",
+                                    sortText: $"{SortingPrefix}{sign}{m.Name}", // Some operators have the same sign, but different methods (e.g. +a and a + b)
+                                    symbols: ImmutableList.Create(m),
+                                    rules: CompletionItemRules.Default,
+                                    contextPosition: position);
+                return operators;
+            }
+
+            return SpecializedCollections.EmptyEnumerable<CompletionItem>();
+        }
+
+        private static bool IsExcludedOperator(IMethodSymbol m)
+        {
+            switch (m.Name)
+            {
+                case WellKnownMemberNames.FalseOperatorName:
+                case WellKnownMemberNames.TrueOperatorName:
+                    return true;
+                default:
+                    return false;
+            }
+        }
+
+        private static bool IsExcludedFromOperators(SemanticModel semanticModel, ITypeSymbol container)
+        {
+            if (container.IsSpecialType())
+            {
+                return true;
+            }
```

Comment:
this needs docs as to why we want thsi.  i actually don't see how this is a good thing.

Reply:
> i actually don't see how this is a good thing.

I removed the exclusion of built-in types for operators, but the result is a mixed bag of types with
* no operators at all (e.g. `int`)
* partial operators (e.g. `string` includes the comparison operators but not the addition/concatenation operator)
* full operators (e.g. `decimal`)

see https://github.com/dotnet/roslyn/pull/47511#discussion_r496959643 for details